sqltree/parser.py

```diff
@@ -2404,6 +2428,17 @@ def _parse_cast_type(p: Parser) -> CastType:
         return kwseq
 
 
+def _maybe_consume_one_of_casttypes(p: Parser) -> Optional[CastType]:
+    casttype = _maybe_parse_cast_type(p)
+    if casttype:
+        if _next_is_stringliteral_type(p):
+            return casttype
+        else:
+            for _ in range(_get_cast_type_keyword_number(casttype)):
```

Comment:
Maybe a more robust solution here would be to save the state of the `p.pi` before we go into this branch and restore it directly, instead of attempting to figure out how many times we have to call `wind_back`. I think it's just an index into the token list.

Reply:
Updated to use `p.pi.next_pos`